controllers/controllers/workloads/integration/cforg_controller_integration_test.go

```diff
@@ -0,0 +1,137 @@
+package integration_test
+
+import (
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/workloads/v1alpha1"
+	"code.cloudfoundry.org/cf-k8s-controllers/controllers/controllers/workloads"
+	. "code.cloudfoundry.org/cf-k8s-controllers/controllers/controllers/workloads/testutils"
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	hncv1alpha2 "sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+var _ = Describe("CFOrgReconciler Integration Tests", func() {
+	const (
+		orgName = "my-org"
+	)
+	var (
+		testCtx       context.Context
+		rootNamespace *v1.Namespace
+		orgGUID       string
+		cfOrg         workloadsv1alpha1.CFOrg
+	)
+
+	BeforeEach(func() {
+		testCtx = context.Background()
+		rootNamespace = createNamespace(testCtx, k8sClient, PrefixedGUID("root-ns"))
+		orgGUID = PrefixedGUID("cf-org")
+		cfOrg = workloadsv1alpha1.CFOrg{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      orgGUID,
+				Namespace: rootNamespace.Name,
+			},
+			Spec: workloadsv1alpha1.CFOrgSpec{
+				Name: orgName,
+			},
+		}
+	})
+
+	AfterEach(func() {
+		Expect(k8sClient.Delete(context.Background(), rootNamespace)).To(Succeed())
+	})
+
+	When("the CFOrg is created", func() {
+		BeforeEach(func() {
+			createNamespace(testCtx, k8sClient, orgGUID)
+			Expect(k8sClient.Create(testCtx, &hncv1alpha2.HierarchyConfiguration{
+				ObjectMeta: metav1.ObjectMeta{Namespace: orgGUID, Name: "hierarchy"},
+			})).To(Succeed())
+		})
+
+		JustBeforeEach(func() {
+			Expect(k8sClient.Create(testCtx, &cfOrg)).To(Succeed())
+		})
+
+		It("creates a subnamespace anchor in the root namespace", func() {
+			var anchor hncv1alpha2.SubnamespaceAnchor
+			Eventually(func() error {
+				return k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &anchor)
+			}).Should(Succeed())
+
+			Expect(anchor.Labels).To(HaveKeyWithValue(workloads.OrgNameLabel, orgName))
+		})
+
+		It("makes the CFOrg the owner of its subnamespace anchor", func() {
+			var anchor hncv1alpha2.SubnamespaceAnchor
+			Eventually(func() error {
+				return k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &anchor)
+			}).Should(Succeed())
+
+			Expect(anchor.ObjectMeta.OwnerReferences).To(ConsistOf([]metav1.OwnerReference{
```

Comment:
FYI, Ginkgo offers an alternative way (see Ginkgo docs [here](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode))to put all the assertions in `Eventually` which from my point of view makes the code slightly more readable. But that is up to you really.

Applies to most of the `Eventually`s in this test.

Reply:
Thanks. Didn't know that. Done